junit-jupiter-api/src/main/java/org/junit/jupiter/api/Assertions.java

```diff
@@ -870,6 +871,38 @@ public static void assertIterableEquals(Iterable<?> expected, Iterable<?> actual
 		AssertIterableEquals.assertIterableEquals(expected, actual, messageSupplier);
 	}
 
+	// --- assertLinesMatch ----------------------------------------------------
+
+	/**
+	 * <em>Asserts</em> that {@code expected} list of {@linkplain String}s match {@code actual}
+	 * list.
+	 *
+	 * <p>This methods differs from other assertions, that effectively only check
+	 * {@link String#equals(Object)}, by using following staged match-making algorithm:
+	 *
+	 * <p>For each pair of expected and actual lines do
+	 * <ol>
+	 *   <li>check if {@code expected.equals(actual)} - if yes, continue with next pair</li>
+	 *   <li>treat {@code expected} as a regular expression and check via {@link String#matches(String)}
+	 *   - if yes, continue with next pair</li>
+	 *   <li>check if {@code expected} is a fast-forward marker, if yes apply fast-forward actual lines
+	 *   accordingly (see below) and goto 1.</li>
+	 * </ol>
+	 *
+	 * <p>A valid fast-forward marker is an expected line that starts and ends with the literal
+	 * {@code >>} and contains at least 4 characters. Examples:
+	 * <ul>
+	 *   <li>{@code >>>>}<br>{@code >> stacktrace >>}<br>{@code >> single line, non Integer.parse()-able comment >>}
+	 *   <br>Skip arbitrary number of actual lines, until first matching subsequent expected line is found. Any
+	 *   character between the fast-forward literals are discarded.</li>
+	 *   <li>{@code ">> 21 >>"}
+	 *   <br>Skip strictly 21 lines. If they can't be skipped for any reason, an error is raised.</li>
+	 * </ul>
+	 */
```

Comment:
I think it's worth giving some examples of how fast-forwarding works (or to provide a link to the tests for this assertion, which themselves provide nice examples IMO).

It's not really clear just from the description what fast-forwarding does, but as I'm not a master of the English language (despite being a native speaker!), I'm struggling to give useful feedback to improve the description itself, hence why I think examples would be useful.